File: gym_client/src/types/payment.ts
export type PaymentMode = 'Cash' | 'Upi' | 'Card'

export interface Payment {
  id: number
  membershipId: number
  amount: number
  paymentDate: string
  paymentMode: PaymentMode
  receiptNo?: string | null
  /** Paid invoice/receipt generated for this transaction */
  invoiceId?: number | null
}

export interface CreatePaymentDto {
  membershipId: number
  amount: number
  paymentDate: string
  paymentMode: PaymentMode
  receiptNo?: string | null
}

export interface UpdatePaymentDto {
  amount?: number | null
  paymentDate?: string | null
  paymentMode?: PaymentMode | null
  receiptNo?: string | null
}

File: src/GymManagement.Domain/Entities/Payment.cs
namespace GymManagement.Domain.Entities
{
    public enum PaymentMode
    {
        Cash,
        Upi,
        Card
    }

    public class Payment : BaseEntity
    {
        public int MembershipId { get; set; }
        public decimal Amount { get; set; }
        public DateTime PaymentDate { get; set; }
        public PaymentMode PaymentMode { get; set; }
        public string? ReceiptNo { get; set; }
        public int? OrganizationId { get; set; }

        public UserMembership Membership { get; set; } = null!;
        public Organization? Organization { get; set; }
        public ICollection<Invoice> Invoices { get; set; } = new List<Invoice>();
    }
}
